types/got/index.d.ts

```diff
@@ -99,6 +88,45 @@ declare namespace got {
         (url: GotUrl, options: GotBodyOptions<null>): GotPromise<Buffer>;
     }
 
+    interface GotJSONFn {
+        (url: GotUrl): GotPromise<any>;
+        (url: GotUrl, options: GotJSONOptions): GotPromise<any>;
+    }
+
+    interface GotFormFn<T extends string | null> {
+        (url: GotUrl): GotPromise<any>;
```

Comment:
This overload should return `Buffer` or `string`. 

Ditto for `GotBodyFn` interface below.

Reply:
I think I've already fixed this, as it's `string` when `T` is `string` and `Buffer` when `T` is `null`